### ISSUE TITLE
Upgrade to openai assistant v2

### ISSUE THREAD

**Issue by lucasdfr** [OvidijusParsiunas/deep-chat#190]
OpenAI has recently released assitant v2, the v1 will be deprecated by the end of 2024.
See : https://platform.openai.com/docs/assistants/migration

**Comment by OvidijusParsiunas** [comment]
Hi @lucasdfr.
Thankyou for letting me know about this! I will start working on it asap!

**Comment by OvidijusParsiunas** [comment]
The OpenAI Assistants v2 integration is now available in `deep-chat-dev` and `deep-chat-react-dev` versions `9.0.173`.

The [Assistants](https://deepchat.dev/docs/directConnection/OpenAI#Assistant) object now has a new property called `files_tool_type` which has the following type:

```files_tool_type?: FileToolTypes | ((fileNames: string[]) => FileToolTypes);```

This uses a new `FileToolTypes` type:

```type FileToolTypes = 'code_interpreter' | 'file_search' | 'images';```

This is basically used to define the tool that will be used when the user uploads files. You can also define a function which returns a particular tool type string based on the uploaded files.
If `files_tool_type` is not defined, Deep Chat automatically defaults to using `images` and simply sends the image to a [_vision_ model](https://platform.openai.com/docs/models) without using the [`code_interpreter`](https://platform.openai.com/docs/assistants/tools/code-interpreter) and [`file_search`](https://platform.openai.com/docs/assistants/tools/file-search) tools.
It is IMPORTANT to note that when [`code_interpreter`](https://platform.openai.com/docs/assistants/tools/code-interpreter) or [`file_search`](https://platform.openai.com/docs/assistants/tools/file-search) are defined, you need to make sure they are toggled on in the [OpenAI Assistant Playground](https://platform.openai.com/playground) or in the [NewAssistant](https://deepchat.dev/docs/directConnection/OpenAI#NewAssistant) object.

Example for defining the `files_tool_type`:

```
directConnection={{
  openAI: {
    assistant: {
      assistant_id: 'assistant with code interpreter',
      files_tool_type: 'code_interpreter',
    },
    key: 'your-key',
  },
}}
```

The [NewAssistant](https://deepchat.dev/docs/directConnection/OpenAI#NewAssistant) object has also been updated to reflect the [New Assistant Open AI API](https://platform.openai.com/docs/api-reference/assistants/createAssistant):

```
type OpenAINewAssistant = {
  model?: string;
  name?: string;
  description?: string;
  instructions?: string;
  tools?: {
    type: 'code_interpreter' | 'file_search' | 'function';
    function?: {name: string; description?: string; parameters?: object};
  }[];
  tool_resources?: {
    code_interpreter?: {
      file_ids: string[];
    };
    file_search?: {
      vector_store_ids?: string[];
      vector_stores: {file_ids: string[]};
    };
  };
}
```

Let me know if you experience any issues using this. Thankyou!

**Comment by OvidijusParsiunas** [comment]
Hi, the functionality mentioned in this issue is now available in Deep Chat version `2.0.0`.
